File: models/user.model.js
module.exports = ({ GenerateID, Connection, DataTypes }) => {
	const UserModel = Connection.define(
		"user",
		{
			user_id: {
				type: DataTypes.STRING(50),
				primaryKey: true,
			},
			name: {
				type: DataTypes.STRING(200),
				allowNull: false,
			},
			email: {
				type: DataTypes.STRING(200),
				allowNull: false,
				unique: true,
			},
			password: {
				type: DataTypes.TEXT,
				allowNull: false,
			},
			is_active: {
				type: DataTypes.BOOLEAN,
				allowNull: false,
				defaultValue: true,
			},
			timestamp: {
				type: DataTypes.BIGINT,
				allowNull: false,
			},
		},
		{
			tableName: "user",
			timestamps: false,
			underscored: true,
			hooks: {
				beforeCreate: (record) => {
					record.user_id = GenerateID("usr");
				},
			},
		}
	);

	return UserModel;
};

File: app/models/user_model.py
from sqlalchemy import Column, String, Text, Boolean, BigInteger
from app.database import Base

class User(Base):
    __tablename__ = "user"  # nama tabel sama seperti di Sequelize

    user_id = Column(String(50), primary_key=True, index=True)
    name = Column(String(100), nullable=False)
    email = Column(String(200), nullable=False, unique=True)
    password = Column(Text, nullable=False)
    is_active = Column(Boolean, nullable=False, default=True)
    timestamp = Column(BigInteger, nullable=False)
